src/components/Table/TableHeader.tsx

```diff
@@ -0,0 +1,63 @@
+import { flexRender, Header } from '@tanstack/react-table';
+import React, { useLayoutEffect, useRef } from 'react';
+import { Flex, Text, Pressable } from 'native-base';
+
+import { Icon } from '../Icon/Icon';
+import { IconNames } from '../../types/icons';
+
+type Props = {
+  header: Header<any, any>;
+  widths: number[];
+  minWidth: number;
+  setWidths: React.Dispatch<React.SetStateAction<number[]>>;
+};
+
+export const TableHeader: React.FC<Props> = ({
+  header,
+  widths,
+  setWidths,
+  minWidth,
+}) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const localWidths = widths;
+
+  useLayoutEffect(() => {
+    const headerWidth = widths[header.index];
+    if (ref?.current?.offsetWidth === headerWidth) {
+      return;
+    }
+
+    localWidths[header.index] = ref?.current?.offsetWidth || headerWidth;
+    setWidths([...localWidths]);
+  }, [ref.current?.offsetWidth]);
+
+  return (
+    <Pressable
+      onPress={header.column.getToggleSortingHandler()}
+      key={header.id}
+      ref={ref}
+    >
+      <Flex
+        flexDirection="row"
+        justifyContent="space-between"
+        alignItems="center"
+        minWidth={`${minWidth}px`}
+      >
+        <Text
+          fontWeight="700"
+          fontSize="xs"
+          lineHeight="4xs"
+          textTransform="capitalize"
+        >
+          {header.isPlaceholder
+            ? null
+            : flexRender(header.column.columnDef.header, header.getContext())}
+        </Text>
+        {{
+          asc: <Icon name={IconNames.SortAsc} pl="8" />,
+          desc: <Icon name={IconNames.SortDesc} pl="8" />,
+        }[header.column.getIsSorted() as string] ?? null}
```

Comment:
Rather than null it'd be good to have an invisible space here so that when the sort icon shows the table doesn't jump around

Reply:
yeah, gotta check with the designers though